### DISCUSSION TITLE
Serenity Browser Auto Download in Docker Container

### DISCUSSION THREAD

**Discussion by pdevaraj14** [serenity-bdd/serenity-core#3293] [Q&A]
Hi, can anyone please tell me if its possible for the selenium webdrivermanager to auto download the browser itself in a linux environment. I'm able to get it locally working on windows but it seems like only the drivers are downloading in the linux image in docker container. The serenity conf file im using is the following: 

`serenity {
    take.screenshots = FOR_EACH_ACTION
}

environment = firefox

environments {
  chrome {
    webdriver {
      autodownload = true
      capabilities {
                     proxy {
                            proxyType = "autodetect"
                      }
        browserName = "chrome"
        browserVersion = "stable"
        acceptInsecureCerts = true
        "goog:chromeOptions" {
          args = ["test-type", "ignore-certificate-errors", "start-maximized", "headless=new"
            "incognito", "disable-infobars", "disable-gpu", "disable-default-apps", "disable-popup-blocking", "disable-dev-shm-usage", "no-sandbox"]
        }
      }
    }
  }
  edge {
    webdriver {
      capabilities {
        browserName = "edge"
        "ms:edgeOptions" {
          args = ["test-type", "ignore-certificate-errors", "headless",
            "incognito", "disable-infobars", "disable-gpu", "disable-default-apps", "disable-popup-blocking"]
        }
      }
    }
  }
  firefox {
    webdriver {
      autodownload = true
      capabilities {
                     proxy {
                            proxyType = "autodetect"
                      }
          browserName = "firefox"
          browserVersion = "118"
          pageLoadStrategy = "normal"
          acceptInsecureCerts = true
          unhandledPromptBehavior = "dismiss"
          "moz:firefoxOptions" {
            args = ["--headless"],
            prefs {
              "javascript.options.showInConsole": false
            },
            log {"level": "info"},
        }
      }
    }
  }
}`

**Comment by wakaleo** [comment]
Selenium will not do that - it is your responsibility to ensure that the correct browser  is installed
